build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle

```diff
@@ -101,7 +101,8 @@ buildScan {
       tag 'CI'
       link 'CI Build', buildKiteUrl
       value 'Job Number', System.getenv('BUILDKITE_BUILD_NUMBER')
-
+      value 'Pipeline', System.getenv('BUILDKITE_PIPELINE_SLUG')
+      value 'Job Name', System.getenv('BUILDKITE_LABEL')
```

Comment:
We should think about sanitizing the label to produce a "slug like" value that's easier to search on. Also stripping out things like emojis (we use them in serverless).